File: 2022年08月/13 769. 最多能完成排序的块/maxChunksToSorted.java
// 执行用时：0 ms, 在所有 Java 提交中击败了100.00%的用户
// 内存消耗：38.9 MB, 在所有 Java 提交中击败了26.99%的用户
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int ans = 0, max = arr[0];
        for (int i = 0; i < arr.length; i++) {
            max = Math.max(max, arr[i]);
            if (max == i)
                ans++;
        }
        return ans;
    }
}

File: src/2022年08月/13 769. 最多能完成排序的块/maxChunksToSorted.js
/**
 * @param {number[]} arr
 * @return {number}
 * 执行用时：64 ms, 在所有 JavaScript 提交中击败了38.33%的用户
 * 内存消耗：40.9 MB, 在所有 JavaScript 提交中击败了61.67%的用户
 */
var maxChunksToSorted = function (arr) {
    let max = arr[0];
    let count = 0;
    for (let i = 0; i < arr.length; i++) {
        max = Math.max(max, arr[i]);
        if (max === i) {
            count++;
        }
    }
    return count;
};